### PR TITLE
Doc: clarifying irrational

### PR DESCRIPTION
**Pull request by PallHaraldsson**
Clarifying (fixing?) #37977

[skip ci]

### CODE REVIEW

File: base/irrationals.jl [hunk 1]
```diff
@@ -5,8 +5,9 @@
 """
     AbstractIrrational <: Real
 
-Number type representing an exact irrational value, which is automatically rounded to the correct precision in
-arithmetic operations with other numeric quantities.
+Number type representing an exact irrational value, which is automatically rounded to the "correct" precision in
+arithmetic operations with other numeric quantities. Even the simplest calculations, like -pi, make convert a
+number to Float64, making the result no longer irrational.
```
Comment (petvana) [inline base/irrationals.jl:10]:
```suggestion
arithmetic operations with other numeric quantities. Even the simplest calculations, like `-pi` or `2pi`, make
convert a number to Float64, making the result no longer irrational.
```

Comment (PallHaraldsson) [reply base/irrationals.jl]: Good suggestion, this is implied multiplication. I did a bit differently and dropped a redundant word.

**Comment by petvana** [comment]
Any specific reason not to format `Float64`, `-pi`, and `2π` as a code? (``` `Float64` ```)

**Comment by PallHaraldsson** [comment]
Not really, I just overlooked that, actually copied from your fixed text (where Float64 wasn't like that, and still the extra word) but then changed to the pi symbol.

File: base/irrationals.jl [hunk 2]
```diff
@@ -5,8 +5,9 @@
 """
     AbstractIrrational <: Real
 
-Number type representing an exact irrational value, which is automatically rounded to the correct precision in
-arithmetic operations with other numeric quantities.
+Number type representing an exact irrational value, which is automatically rounded to the "correct" precision in
+arithmetic operations with other numeric quantities. Even the simplest calculations, such as `-pi` or `2π` convert a
+number to `Float64`, making the result no longer irrational.
```
Comment (ViralBShah) [inline base/irrationals.jl:10]: Maybe rephrase:

Calculations involving arithmetic between numeric types and irrational types will lead to conversion to a numeric type. For example, -pi or 2π produce `Float64` results.

Comment (petvana) [reply base/irrationals.jl:10]: I would like to note that `-pi` is unary, and thus not an arithmetic operation between a numeric type and irrational.

Comment (ViralBShah) [reply base/irrationals.jl:10]: Maybe remove it then?

Comment (PallHaraldsson) [reply base/irrationals.jl:10]: I was jus going o change, but I have second thoughts about "Calculations involving arithmetic" (as synonyms?) and "between numeric types". With pi a number, this could be misunderstood, so I was thinking `Number` types, but I see it has misleading help "Abstract supertype for all number types".

Also should clarification rather go into Irrational and/or AbstractIrrational?

Comment (PallHaraldsson) [reply base/irrationals.jl:10]: "-pi is unary, and thus not an arithmetic operation", mathematically not an operation? But in Julia, it's a unary operator as in almost all languages, and involves calculation:

-(x::AbstractIrrational) = -Float64(x)
